### ISSUE TITLE
hotfixes inconsistently applied on new mirror

### ISSUE THREAD

**Issue by minrk** [conda-forge/conda-forge-repodata-patches-feedstock#2]
Hotfixes from this repo seem to be being applied inconsistently with the new anaconda.org mirror.

For instance, try using the new mirror (this matches what's happening on conda-forge CI):

    conda create --debug -vv --override-channels -c https://conda-static.anaconda.org/conda-forge --dry-run -p /tmp/... petsc=3.10.2

and it will fail with:

```
conda.exceptions.PackagesNotFoundError: The following packages are not available from current channels:

  - petsc=3.10.2
  - *[track_features=blas_openblas]
```

But if you install from the old url (this matches what I get with `-c conda-forge` on my laptop):

```
conda create --debug -vv --override-channels -c https://conda.anaconda.org/conda-forge --dry-run -p /tmp/... petsc=3.10.2
```

it installs just fine. @soapy1 pointed out that this is due to the patches from this repo, which explicitly remove `track_features: blas_openblas` from the blas mpkg.

I discovered that this is related to a mistake I made long ago in a few packages, adding the `blas_openblas` feature. I later learned that this is an irreversible mistake, because if any release has blas_openblas, all future releases that remove this erroneously added feature until blas stops tracking this feature.

The hotfixes here remove the track_features from the blas metapackage (yay!) and the `features` from all packages at the time of the hotfix, but there are two problems:

1. new builds since the last hotfix rendering have the feature (I can fix this by stopping new builds with the feature, or the repo can rebuild the hotfixes which will fix the affected packages)
2. the hotfixes are only applied on the new mirror, not the old url, so I still have the problem *sometimes* but not all the time.

So it seems that the hotfixes *should* mean that I can remove the feature where I had added it (yay!) and solve the problem going forward, but the problem I have right now is that when I install locally it's not using the hotfixes so the `blas` package still tracks this feature, meaning that I will still get the latest package with the feature rather than the latest package.

Is the fix to remove the feature from all my packages and wait for the hotfixes to propagate? Is it intentional that the hotfixes applied on conda-static.anaconda.org are not the same as conda.anaconda.org?

**Comment by soapy1** [comment]
> Is it intentional that the hotfixes applied on conda-static.anaconda.org are not the same as conda.anaconda.org

Hot fixes are only applied to the mirror, that would be conda-static. And conda.anaconda.org should be sending you stuff from conda-static.anaconda.org for the conda-forge channel. Hotfixes are applied whenever an index is run, so that is every 20 minutes.

**Comment by minrk** [comment]
As requested on gitter, the exact commands I'm running are above. and my condarc:

```yaml
channels:
  - conda-forge
  - defaults
show_channel_urls: true
```

**Comment by minrk** [comment]
From gitter: it appears sensitive to whether the user is authenticated, in which case urls are `/t/<token>/channel/...` instead of `/channel/...`.

**Comment by minrk** [comment]
It looks like the gcc7 label packages aren't patched either, only the main label

**Comment by isuruf** [comment]
@minrk, can you bump the version in this repo so that this can be fixed temporarily?

**Comment by soapy1** [comment]
@minrk I was trying to reproduce your issue with
```
$ conda create --debug -vv --override-channels -c https://conda.anaconda.org/conda-forge --dry-run -p /tmp/... petsc=3.10.2
```
and I still get an error
```
conda.exceptions.PackagesNotFoundError: The following packages are not available from current channels:

  - petsc=3.10.2 -> *[track_features=blas_openblas]
```

did you happen to just build this package? Maybe this inconsistency is happening because the clone did not happen until recently?

**Comment by minrk** [comment]
@soapy1 petsc was built several days ago, but this is also affecting all packages with `blas_openblas` feature since ~August when the hotfix package was last built. For instance, try to install numpy:

    conda install -c conda-forge numpy

You should get the latest, which is 1.15.4, but instead you will get 1.15.1, the last version before the hotfix was applied.

#3 fixes this by reverting the blas_openblas feature patches.

**Comment by minrk** [comment]
This appears to be fixed now! After #3 has been merged, I can install numpy 1.15.4 and petsc 3.10.2.

**Comment by msarahan** [comment]
> new builds since the last hotfix rendering have the feature (I can fix this by stopping new builds with the feature, or the repo can rebuild the hotfixes which will fix the affected packages)

Yes, this is an unfortunate side-effect of how the patch data gets generated.  We can't run arbitrary user code on our servers, so rather than the patch being a set of logic (python file) that acts dynamically on repodata, patch data for conda-forge has to be a plain JSON set of replacement content.  Since that replacement content is based on a snapshot of the repo at the time that the replacement content is generated, it does not apply to new packages.

One way around this is to automatically regenerate this feedstock periodically, so that the patches get generated for new content.
